src/system/SystemPacketBuffer.cpp

```diff
@@ -333,14 +333,18 @@ PacketBuffer* PacketBuffer::Consume(uint16_t aConsumeLength)
 {
     PacketBuffer* lPacket = this;
 
-    while (lPacket != NULL && aConsumeLength > 0)
+    while (aConsumeLength > 0)
```

Comment:
Removing the NULL check causes a potential fault on line 338 if lPacket is NULL. Is the argument that the `this` pointer should never be NULL?